routes/apiv1.js

```diff
@@ -76,9 +76,19 @@ router.post('/login', function (req, res, next) {
 	});
 });
 
-router.use(passport.authenticate('jwt', {
-	session: false
-}));
+router.use(function (req, res, next) {
+	passport.authenticate('jwt', {
+		session: false
+	}, function (err, user, info) {
```

Comment:
Passport is kinda stupid or we are but anyway,
1 - err is not null only in case of an internal error or something or with mongo for example, you define this in authenticator line 17.
2 - Errors from JWT passport are passed as error objects in `info`.
3 - if `JWT` token is invalid info will contain an error raised from the strategy, the error is not serializable thus you need to use toString on it.
4 - `Invalid Credentials` will occur only if the user is not found in the database, so you need to pass it as `new error()` to be like point 3.